socorro/external/fs/crashstorage.py

```diff
@@ -663,68 +659,115 @@ class TarFileCrashStore(CrashStorageBase):
-            self.tar_file = self.tarfile_module.open(
-                self.config.tarball_name,
-                'w'
-            )
-            self.tar_file.add(actual_pathname, target_pathname)
+        self.tar_fp = self._create_tarfile()
 
+    #------------------------------------------------------------------------------
     def close(self):
-        try:
-            self.tar_file.close()
-        except AttributeError:
-            # the tar_file was never actually created because the save_*
-            # were never called.  we can silently ignore this
-            pass
+        self.tar_fp.close()
 
+    #------------------------------------------------------------------------------
     def save_processed(self, processed_crash):
         processed_crash_as_string = json.dumps(
             processed_crash,
             default=self.stringify_datetimes
         )
-        crash_id = processed_crash['crash_id']
-        file_name = os.path.join(
-            self.config.temp_directory,
-            crash_id + '.jsonz'
+        crash_id = processed_crash["crash_id"]
+
+        compressed_crash = StringIO()
+        gzip_file = self.gzip_module.GzipFile(fileobj=compressed_crash, mode='w')
+        gzip_file.write(processed_crash_as_string)
+        gzip_file.close()
+        compressed_crash.seek(0)
+        tarinfo = self.tarfile_module.TarInfo('%s.jsonz' % crash_id)
+        tarinfo.size = len(compressed_crash.getvalue())
+        self.tar_fp.addfile(tarinfo, compressed_crash)
+        self.config.logger.debug(
+            'TarFileCrashStore saved - %s to %s',
+            crash_id,
+            self.config.tarball_name
         )
-        file_handle = self.gzip_module.open(file_name, 'w', 9)
-        try:
-            file_handle.write(processed_crash_as_string)
-        finally:
-            file_handle.close()
-        self._save_to_tarfile(
-            file_name,
-            os.path.join(
-                crash_id[:2],
-                crash_id[2:4],
-                crash_id + '.jsonz'
-            )
+
+
+#==============================================================================
+class TarFileSequentialReadingCrashStore(CrashStorageBase):
+    required_config = Namespace()
+    required_config.add_option(
+        name='tarball_name',
+        doc='pathname to a the target tarfile',
+        default='fred.tar'
+    )
+    required_config.add_option(
+        name='tarfile_module',
+        doc='a module that supplies the tarfile interface',
+        default='tarfile',
+        from_string_converter=class_converter
+    )
+    required_config.add_option(
+        name='gzip_module',
+        doc='a module that supplies the gzip interface',
+        default='gzip',
+        from_string_converter=class_converter
+    )
+
+    #------------------------------------------------------------------------------
+    @staticmethod
+    def stringify_datetimes(obj):
+        if isinstance(obj, datetime.datetime):
```

Comment:
If you change this to `datetime.date` the if statement would be true for **both** `datetime.date` **and** `datetime.datetime` objects.